app/javascript/article-form/components/ArticleCoverImage.jsx

```diff
@@ -10,6 +10,51 @@ import { Button } from '@crayons';
 import { Spinner } from '@crayons/Spinner/Spinner';
 import { DragAndDropZone } from '@utilities/dragAndDrop';
 
+const NativeIosImageUpload = ({
+  extraProps,
+  uploadLabel,
+  isUploadingImage,
+  handleNativeMessage,
+}) =>
+  isUploadingImage ? null : (
+    <Fragment>
+      <Button
+        variant="outlined"
+        className="mr-2 whitespace-nowrap"
+        {...extraProps}
+      >
+        {uploadLabel}
+      </Button>
+      <input
+        type="hidden"
+        id="native-cover-image-upload-message"
+        value=""
+        onChange={handleNativeMessage}
+      />
+    </Fragment>
+  );
```

Comment:
I was unable to upload the Cover Image using the native app integration, and I think the problem comes from rendering `null` here when the image is uploading.

This input field (`#native-cover-image-upload-message`) is the entrypoint for native messages incoming to the DOM. So when the first message arrives ("uploading an image") it makes this fragment go away, and then the second message ("success + uploaded image URL" or "failed") fails to be received because the input is no longer there.

I think we would need to move this input somewhere else so it always remains in the DOM regardless if we're uploading or not. Let me know how this sounds @aitchiss

Reply:
Good spot! I've changed it up so we always have the hidden input on the page, and we just hide the button when it's uploading. Seems to work in the browser on the simulator, but let me know how it looks to you 👍